internal/provider/website_resource.go

```diff
@@ -39,63 +39,82 @@ func (r *websiteResource) Configure(ctx context.Context, req resource.ConfigureR
 	r.client = client
 }
 
-func (r *websiteResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
-	var tfPlan websiteResourceModel
-	resp.Diagnostics.Append(req.Plan.Get(ctx, &tfPlan)...)
-	if resp.Diagnostics.HasError() {
-		return
-	}
+func getAvailabilityOrRum(monitoring *websiteMonitoring) (*swoClient.AvailabilityCheckSettingsInput, *swoClient.RumMonitoringInput) {
 
-	// Create our input request.
 	var checkForString *swoClient.CheckForStringInput
-	if tfPlan.Monitoring.Availability.CheckForString != nil {
+	if monitoring.Availability.CheckForString != nil {
 		checkForString = &swoClient.CheckForStringInput{
-			Operator: swoClient.CheckStringOperator(tfPlan.Monitoring.Availability.CheckForString.Operator.ValueString()),
-			Value:    tfPlan.Monitoring.Availability.CheckForString.Value.ValueString(),
+			Operator: swoClient.CheckStringOperator(monitoring.Availability.CheckForString.Operator.ValueString()),
+			Value:    monitoring.Availability.CheckForString.Value.ValueString(),
```

Comment:
The whole checkForString declaration and if block should be moved into the if block checking that monitoring.Availability is not nil a few lines down.